codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/RetryConfigDecorator.kt

```diff
@@ -0,0 +1,156 @@
+
+/* Example Generated Code */
+/*
+pub struct Config {
+    pub(crate) retry_config: Option<smithy_types::retry::RetryConfig>,
+}
+
+impl std::fmt::Debug for Config {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let mut config = f.debug_struct("Config");
+        config.finish()
+    }
+}
+
+impl Config {
+    pub fn builder() -> Builder {
+        Builder::default()
+    }
+}
+
+#[derive(Default)]
+pub struct Builder {
+    retry_config: Option<smithy_types::retry::RetryConfig>,
+}
+
+impl Builder {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    pub fn retry_config(
+        mut self,
+        retry_config: impl Into<Option<smithy_types::retry::RetryConfig>>,
+    ) -> Self {
+        self.retry_config = retry_config.into();
+        self
+    }
+
+    pub fn build(self) -> Config {
+        Config {
+            retry_config: self.retry_config,
+        }
+    }
+}
+
+#[test]
+fn test_1() {
+    fn assert_send_sync<T: Send + Sync>() {}
+    assert_send_sync::<Config>();
+}
+ */
+
+class RetryConfigDecorator : RustCodegenDecorator {
+    override val name: String = "RetryConfig"
+    override val order: Byte = 0
+
+    override fun configCustomizations(
+        protocolConfig: ProtocolConfig,
+        baseCustomizations: List<ConfigCustomization>
+    ): List<ConfigCustomization> {
+        return baseCustomizations + RetryConfigProviderConfig(protocolConfig.runtimeConfig)
+    }
+
+    override fun operationCustomizations(
+        protocolConfig: ProtocolConfig,
+        operation: OperationShape,
+        baseCustomizations: List<OperationCustomization>
+    ): List<OperationCustomization> {
+        return baseCustomizations + RetryConfigConfigPlugin()
+    }
+
+    override fun libRsCustomizations(
+        protocolConfig: ProtocolConfig,
+        baseCustomizations: List<LibRsCustomization>
+    ): List<LibRsCustomization> {
+        return baseCustomizations + PubUseRetryConfig(protocolConfig.runtimeConfig)
+    }
+}
+
+class RetryConfigProviderConfig(runtimeConfig: RuntimeConfig) : ConfigCustomization() {
+    private val retryConfig = smithyTypesRetry(runtimeConfig)
+    private val codegenScope = arrayOf("RetryConfig" to retryConfig.member("RetryConfig"))
+    override fun section(section: ServiceConfig) = writable {
+        when (section) {
+            is ServiceConfig.ConfigStruct -> rustTemplate("pub(crate) retry_config: Option<#{RetryConfig}>,", *codegenScope)
+            is ServiceConfig.ConfigImpl -> emptySection
+            is ServiceConfig.BuilderStruct ->
+                rustTemplate("retry_config: Option<#{RetryConfig}>,", *codegenScope)
+            ServiceConfig.BuilderImpl ->
+                rustTemplate(
+                    """
+            pub fn retry_config(mut self, retry_config: impl Into<Option<#{RetryConfig}>>) -> Self {
+                self.retry_config = retry_config.into();
+                self
+            }
+            """,
```

Comment:
I still need to get around to writing a pre-commit script that corrects the indentation on these 😅 

Should this builder be consistent with the one in `aws-types`?